src/common/xo-week-charts/index.js

```diff
@@ -0,0 +1,414 @@
+
+    // Outside limits.
+    if (d1 === undefined) {
+      return
+    }
+
+    const elem = date - d0.date > d1.date - date ? d1 : d0
+    const x = this.x(elem.date)
+
+    const { props } = this
+    const hover = this.svg.select('.hover-container')
+
+    hover
+      .select('.hover-line')
+        .attr('x1', x)
+        .attr('x2', x)
+
+    hover
+      .select('.hover-text')
+        .attr('dx', x + 5)
+        .attr('dy', props.chartHeight / 2)
+        .text(props.valueRenderer(elem.value))
+  }
+
+  componentDidMount () {
+    // Horizon area ----------------------------------------
+
+    const svg = this.svg = d3
+      .select(this.refs.chart)
+        .append('svg')
+          .attr('transform', `translate(${HORIZON_AREA_MARGIN}, 0)`)
+
+    svg.append('g').attr('class', 'x-axis').style(X_AXIS_STYLE)
+    svg.append('g').attr('class', 'horizon-area')
+    svg.append('text').attr('class', 'label').style(LABEL_STYLE)
+
+    this.x = d3.time.scale()
+    this.y = d3.scale.linear()
+
+    this.bisectDate = d3.bisector(elem => elem.date).left
+
+    this.xAxis = d3
+      .svg
+      .axis()
+      .scale(this.x)
+      .orient('bottom')
+
+    this.line = d3
+      .svg
+      .line()
+      .x(elem => this.x(elem.date))
+      .y(elem => this.y(elem.value))
+
+    // Tooltip ---------------------------------------------
+
+    svg
+      .append('rect')
+        .attr('class', 'mouse-area')
+        .style(MOUSE_AREA_STYLE)
+        .on('mousemove', this._handleMouseMove)
+
+    const hover = svg
+      .append('g')
+        .attr('class', 'hover-container')
+        .style('pointer-events', 'none')
+
+    hover
+      .append('line')
+        .attr('class', 'hover-line')
+        .attr('y1', 0)
+        .style(HOVER_LINE_STYLE)
+
+    hover
+      .append('text')
+        .attr('class', 'hover-text')
+        .style(HOVER_TEXT_STYLE)
+
+    this._draw()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { props } = this
+
+    if (!propsEqual(
+      props,
+      nextProps,
+      [ 'chartHeight', 'chartWidth', 'data', 'maxValue' ]
+    )) {
+      this._draw(nextProps)
+    }
+
+    if (props.tooltipX !== nextProps.tooltipX) {
+      this._updateTooltip(nextProps.tooltipX)
+    }
+  }
+
+  render () {
+    return <div ref='chart' />
+  }
+}
+
+// ===================================================================
+
+@propTypes({
+  chartHeight: propTypes.number,
+  series: propTypes.arrayOf(
+    propTypes.shape({
+      data: propTypes.arrayOf(
+        propTypes.shape({
+          date: propTypes.number.isRequired,
+          value: propTypes.number.isRequired
+        })
+      ).isRequired,
+      objectId: propTypes.string.isRequired
+    })
+  ).isRequired,
+  valueRenderer: propTypes.func
+})
+export default class XoWeekCharts extends Component {
+  _handleResize = () => {
+    const { container } = this.refs
+    this.setState({
+      chartsWidth: container && container.offsetWidth || undefined
```

Comment:
It's means to use the default prop ```chartsWidth``` of ```XoWeekChart```.
Without ```|| undefined```, ```chartsWidth``` can be equals to false. It's not a good value.